### PR TITLE
Update references to fingerprinted assets in sources

### LINKED ISSUE #5: Support recursive asset fingerprinting
I tried to fingerprint an image URL in my stylesheet, which itself is pulled in by index.html, only to find that the fingerprinted CSS doesn't contain the fingerprinted image URL.

This is easy to reproduce by augmenting the example project slightly by changing `app.css` to:

``` css
body {
    background-image: url('${img/foo.png}');
    color: red;
}
```

And of course adding some file at `img/foo.png`. Now, in the target dir `app.css` has the fingerprinted path:

``` css
body {
    background-image: url('/img/foo-3a6b2cc87905e826f9b5e8079fc5b74e.png');
    color: red;
}
```
But the fingerprinted CSS file `app-3d31d9b4931c0cfd1054f314aa884d3b.css `, being referred to in the `index.html`, does not.

I assume that some sort of topological sort is needed to process the top level files first, and then the first level fingerprinted files, and so on. I haven't looked at the code in detail, but I'd be happy to have a stab at solving this.

Thanks!

### PR DESCRIPTION
**Pull request by lsnape**
Fixes #5 

The implementation was easier than I thought. However, I have deliberately kept the change as small as possible. Feedback welcome :)

### CODE REVIEW

File: src/afrey/boot_asset_fingerprint.clj [hunk 1]
```diff
@@ -67,6 +75,7 @@
                                        (:path)
                                        (get file-rename-hash)
                                        (io/file output-dir))]]
+                      :when (not (contains? (set (map :path sources)) (:path file)))]
```
Comment (AdamFrey) [inline src/afrey/boot_asset_fingerprint.clj]: You've got an unbalanced square bracket here. You want to delete the last bracket on the line above, I believe

**Review by AdamFrey — CHANGES_REQUESTED** [review]

**Comment by AdamFrey** [comment]
Thanks for this PR! I'm looking over it now

**Comment by lsnape** [comment]
Sorry about that. Looks like I was a bit overzealous with discarding the hunks pre-commit!

**Comment by AdamFrey** [comment]
I added your commit, did some work on top of it and pushed it up as 1.4.0. Please check it out and let me know if everything looks good for you.

**Pull request closed without merging**